### PR TITLE
[Core] Improve handling of empty library path strings

### LINKED ISSUE #26: `BAL_LIBRARY_PATH` ignored
## What

Since we implemented the settings mechanism in BAL, we accidentally cause init to ignore the `BAL_LIBRARY_PATH` env var.

## Expected behaviour

When BAL is initialized, if the library path is set to an empty string, then it should follow the same initialization logic as if it was not set at all.

### PR DESCRIPTION
**Pull request by foundrytom**
Fallback to BAL_LIBRARY_PATH was inadvertently broken when we added the settings mechanism. This is due to the fact we never considered a setting set to an empty string as being "unset".

This improves behaviour such that if the `library_path` setting is set, but to an emtpy string, we still fall back on the env var. Only then, if we have an empty string/None, then we grumble.

This rejigs the logic a little in init to also ensure we don't update the settings in the manager if we raise an exception due to library path being empty. This makes things more atomic, which is a good thing.

NB. This bumps the required OpenAssetIO version to alpha-9 for tests.

Closes #26

### CODE REVIEW

File: plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py [hunk 1]
```diff
@@ -72,28 +72,33 @@ def settings(self, hostSession):
...
+        # existing path.
+        existing_library_path = self.__settings.get("library_path")
+        library_path = managerSettings.get("library_path", existing_library_path)
 
-        if self.__settings.get("library_path") is None:
+        if not library_path:
             hostSession.logger().log(
                 hostSession.logger().Severity.kDebug,
-                f"'library_path' not in settings, checking {self.__lib_path_envvar_name}",
+                f"'library_path' not in settings or empty, checking {self.__lib_path_envvar_name}",
```
Comment (elliotcmorris) [inline plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py:84]: [Consider][Minor]
`not in settings or empty` is slightly ambiguous, as in, NOT (in setting || empty) or NOT (in settings) || empty,  `not in settings or is empty` clarifies to my eye.

**Review by elliotcmorris — APPROVED** [review]
Fabulous. Dabulous. Crabulous :crab:

File: plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py (hunk 1 above)
Comment (foundrytom) [reply plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py]: Good call, fixed [here](https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/27/commits/9ce7f5cfd83bfa4bbbdee84148d2853e926b6263).

File: plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py [hunk 2]
```diff
@@ -72,28 +72,33 @@ def settings(self, hostSession):
...
             )
-            self.__settings["library_path"] = os.environ.get(self.__lib_path_envvar_name)
+            library_path = os.environ.get(self.__lib_path_envvar_name)
+
+        if not library_path:
+            raise PluginError(f"'library_path'/{self.__lib_path_envvar_name} not set or empty")
```
[the remark this answers is not in the document]
Comment (foundrytom) [reply plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py]: Good call, fixed [here](https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/27/commits/9ce7f5cfd83bfa4bbbdee84148d2853e926b6263).

**Review by feltech — APPROVED** [review]
LGTM squash and merge when CI passing

**Pull request merged**